### PR TITLE
Rollup of 24 pull requests

### PR DESCRIPTION
**Pull request by alexcrichton**
- Successful merges: #37255, #37317, #37408, #37410, #37422, #37427, #37470, #37501, #37537, #37556, #37557, #37564, #37565, #37566, #37569, #37574, #37577, #37579, #37583, #37585, #37586, #37587, #37589, #37596
- Failed merges: #37521, #37547

### CODE REVIEW

File: src/test/compile-fail/windows-subsystem-gated.rs [hunk 1]
```diff
@@ -9,6 +9,6 @@
 // except according to those terms.
 
 #![windows_subsystem = "console"]
-//~^ ERROR: the windows subsystem attribute is currently unstable
```
Comment (TimNN) [inline src/test/compile-fail/windows-subsystem-gated.rs]: Shouldn't the fix rather updated the emitted error message?

**Comment by alexcrichton** [comment]
Oops, right!

**Comment by TimNN** [comment]
🎉

**Pull request merged**